lib/Subscriber/Behavior/Path/ExplicitSubscriber.php

```diff
@@ -141,36 +138,74 @@ public function handlePersist(PersistEvent $event)
         $event->setNode($node);
     }
 
-    private function renameNode(NodeInterface $node, $nodeName)
+    /**
+     * Handle the existing node.
+     *
+     * If the path of the parent node in the event is different from the path
+     * of the actual parent node path for the document then we perform an
+     * implicit move.
+     *
+     * If the node name is different from the of the PHPCR node, then we rename
+     * the PHPCR node accordingly.
+     *
+     * Otherwise, there is nothing to do.
+     *
+     * @param NodeManager $nodeManager
+     * @param NodeInterface $node  The existing node.
+     * @param NodeInterface $parentNode  The parent node from the event (not necessarily the subject node)
+     * @param string $nodeName  The target name of the node.
+     */
+    private function handleExisting(NodeManager $nodeManager, NodeInterface $node, NodeInterface $parentNode, $nodeName)
     {
+        $realParentPath = PathHelper::getParentPath($node->getPath());
+        $eventParentPath = $parentNode->getPath();
+
+        if ($eventParentPath !== $realParentPath) {
```

Comment:
I actually have no idea why this method was added.

On Wed, Apr 27, 2016 at 11:13:40PM -0700, Daniel Rotter wrote:
>    In [1]lib/Subscriber/Behavior/Path/ExplicitSubscriber.php:
> 
>  >      {
>  > +        $realParentPath = PathHelper::getParentPath($node->getPath());
>  > +        $eventParentPath = $parentNode->getPath();
>  > +
>  > +        if ($eventParentPath !== $realParentPath) {
> 
>    Shouldn't changing of the parent path be handled by the ParentSubscriber?
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu/sulu-document-manager/pull/66#discussion_r61377583
>    2. https://github.com/sulu/sulu-document-manager/pull/66/files/05556aabdb564b020fa388baf338b54d39353978#r61377583
>    3. https://github.com/sulu/sulu-document-manager/pull/66/files/05556aabdb564b020fa388baf338b54d39353978#r61377583